src/Engine/Manager/SoundManager.cpp

```diff
@@ -114,3 +113,43 @@ void SoundManager::Update(World& world) {
         break;
     }
 }
+
+Component::SoundSource* SoundManager::CreateSoundSource() {
+    return soundSources.Create();
+}
+
+Component::SoundSource* SoundManager::CreateSoundSource(const Json::Value& node) {
+    Component::SoundSource* soundSource = soundSources.Create();
+    
+    // Load values from Json node.
+    std::string name = node.get("sound", "").asString();
+    if (!name.empty())
+        soundSource->soundBuffer = Managers().resourceManager->CreateSound(name);
+    
+    soundSource->pitch = node.get("pitch", 1.f).asFloat();
+    soundSource->gain = node.get("gain", 1.f).asFloat();
+    soundSource->loop = node.get("loop", false).asBool();
+    
+    return soundSource;
+}
+
+const std::vector<Component::SoundSource*>& SoundManager::GetSoundSources() const {
+    return soundSources.GetAll();
+}
+
+Component::Listener* SoundManager::CreateListener() {
+    return listeners.Create();
+}
+
+Component::Listener* SoundManager::CreateListener(const Json::Value& node) {
+    return listeners.Create();
```

Comment:
Don't know if there's anything to load for `Component::Listener`, just putting this here in case something was overlooked.

Reply:
Listener doesn't actually have any properties.